scalafix-core/shared/src/main/scala/scalafix/internal/rule/RuleCtxImpl.scala

```diff
@@ -87,6 +89,51 @@ case class RuleCtxImpl(tree: Tree, config: ScalafixConfig) extends RuleCtx {
   def addLeft(tok: Token, toAdd: String): Patch = Add(tok, toAdd, "")
   def addLeft(tree: Tree, toAdd: String): Patch =
     toks(tree).headOption.fold(Patch.empty)(addLeft(_, toAdd))
+  def trimLeft(token: Token): Patch = trimLeft(token, dropFirstNewline = true)
+  def trimSpacesLeft(token: Token): Patch =
+    trimLeft(token, dropFirstNewline = false)
+  private def trimLeft(token: Token, dropFirstNewline: Boolean): Patch = {
+    val builder = Seq.newBuilder[Token]
+    val spaces = tokenList.leading(token).takeWhile(_.is[Space])
+    builder ++= spaces
+
+    if (dropFirstNewline) {
+      val tokenBeforePotentialNewline = spaces.lastOption.getOrElse(token)
+      val prev1 = tokenList.prev(tokenBeforePotentialNewline)
+      val prev2 = tokenList.prev(prev1)
+      val prev3 = tokenList.prev(prev2)
+
+      if (isWindowsNewline(prev2, prev1)) { // flip args as leading tokens are in reverse order
+        if (!isSingleLineComment(prev3)) builder += (prev1, prev2)
+      } else if (prev1.is[Newline] && !isSingleLineComment(prev2))
+        builder += prev1
+    }
+    removeTokens(builder.result())
+  }
+  private def isSingleLineComment(tk: Token): Boolean =
+    tk.is[Comment] && tk.text.startsWith("//")
+  private def isWindowsNewline(tk1: Token, tk2: Token): Boolean =
+    tk1.is[CR] && tk2.is[LF]
```

Comment:
I ran tests on Windows with a file containing CR+LF characters and it seems that scalameta ignore the CR ones, keeping only the LF tokens

Reply:
We have some windows line break related issues that I have never properly looked into https://github.com/scalameta/scalameta/issues/443